packages/backend/src/open_payments/payment/incoming/model.ts

```diff
@@ -76,6 +76,7 @@ export class IncomingPayment
 
   public readonly assetId!: string
   public asset!: Asset
+  public receivedAmount?: Amount
```

Comment:
You could make `receivedAmount` required by adding it and `receivedAmountValue` as [virtual attributes](https://github.com/interledger/rafiki/blob/b038caaa90063164d0a0f02515cdeb59b7170844/packages/backend/src/open_payments/payment/incoming/model.ts#L53-L54)
```suggestion
  private receivedAmountValue?: bigint

  public get receivedAmount(): Amount {
    return {
      value: this.receivedAmountValue || BigInt(0),
      assetCode: this.asset.code,
      assetScale: this.asset.scale
    }
  }

  public set receivedAmount(amount: Amount) {
    this.receivedAmountValue = amount.value
  }
```

Reply:
Should I then also add it to postgres?

Reply:
No this let's the model interface (and serialized JSON) include `receivedAmount` without it existing in the postgres table:
https://vincit.github.io/objection.js/api/model/static-properties.html#static-virtualattributes